docs/architecture/distributed-cloud-native-apps-containers/event-based-communication-patterns/background-tasks-with-ihostedservice.md.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Implement background tasks in microservices with IHostedService and the BackgroundService class
+description: .NET Microservices Architecture for Containerized .NET Applications | Understand the new options to use IHostedService and BackgroundService to implement background tasks in microservices .NET Core.
+ms.date: 01/13/2021
+---
+# Implement background tasks in microservices with IHostedService and the BackgroundService class
+
+[!INCLUDE [download-alert](../includes/download-alert.md)]
+
+In the world of microservices, background tasks are essential for operations that need to run independently of user interactions, such as data processing, batch jobs, or periodic maintenance. 
+
+Background tasks and scheduled jobs are something you might need to use in any application, whether or not it follows the microservices architecture pattern. The difference when using a microservices architecture is that you can implement the background task in a separate process/container for hosting so you can scale it down/up based on your need.
+
+From a generic point of view, in .NET we called these type of tasks *Hosted Services*, because they are services/logic that you host within your host/application/microservice. Note that in this case, the hosted service simply means a class with the background task logic.
+
+Since .NET Core 2.0, the framework provides a new interface named <xref:Microsoft.Extensions.Hosting.IHostedService> helping you to easily implement hosted services. The basic idea is that you can register multiple background tasks (hosted services) that run in the background while your web host or host is running, as shown in the image 7-26.
+
+![Diagram comparing ASP.NET Core IWebHost and .NET Core IHost.](./media/ihosted-service-webhost-vs-host.png)
+
+**Figure 7-26**. Using IHostedService in a WebHost vs. a Host
+
+ASP.NET Core 1.x and 2.x support `IWebHost` for background processes in web apps. .NET Core 2.1 and later versions support `IHost` for background processes with plain console apps. Note the difference made between `WebHost` and `Host`.
```

Comment:
```suggestion
**Figure 7-26**. Using IHostedService in a WebHost vs. a Host

ASP.NET Core 1.x and 2.x support `IWebHost` for background processes in web apps. .NET Core 2.1 and later versions support `IHost` for background processes with plain console apps. Note the difference made between `WebHost` and `Host`.
```

ASP.NET Core 1.x and 2.x have been out of support for many years now. I would recommend updating this and the following section significantly, assuming .NET Core 2.1 + and if anything just adding a note about IWebHost.